src/utils/product-utils.ts

```diff
@@ -43,12 +43,30 @@ export function getSKU(
 
 export function getSelectedVariant(
     product: Product | SerializeFrom<Product>,
-    selectedChoices: Record<string, wixStoresProducts.Choice | undefined> = {}
+    selectedChoices: Record<string, wixStoresProducts.Choice | undefined>
 ): wixStoresProducts.Variant | undefined {
     const selectedChoiceValues = selectedChoicesToVariantChoices(product, selectedChoices);
     return product.variants?.find((variant) => deepEqual(variant.choices, selectedChoiceValues));
 }
 
+export function getMatchingVariants(
+    product: Product | SerializeFrom<Product>,
+    selectedChoices: Record<string, wixStoresProducts.Choice | undefined>
+): wixStoresProducts.Variant[] {
+    const selectedChoiceValues = selectedChoicesToVariantChoices(product, selectedChoices, {
```

Comment:
let's just filter out attributes with `undefined` values here and not introduce the `option` parameter to the `selectedChoicesToVariantChoices`.